contrib/install.sh

```diff
@@ -0,0 +1,1365 @@
+        for pkg in "${packages[@]}"; do
+            if dpkg -l | grep -q "^ii  $pkg "; then
+                to_remove+=("$pkg")
+            fi
+        done
+
+        # Remove packages if any
+        if [ ${#to_remove[@]} -gt 0 ]; then
+            echo "🧹 Removing packages: ${to_remove[*]}"
+            sudo apt-get remove -y --purge "${to_remove[@]}"
+            sudo apt-get autoremove -y
+            sudo apt-get clean -y
+        fi
+    fi
+    echo "✅ Aptget cleanup complete!"
+}
+
+# Rust cleanup
+# Remove toolchai and directories built within
+# $HOME/.cargo/bin/rustup
+cleanup_rust() {
+    echo "🧹 Cleaning up Rust toolchain"
+    if command -v rustup >/dev/null 2>&1; then
+        echo 'y' | rustup self uninstall
+    elif [ -f "$HOME/.cargo/bin/rustup" ]; then
+        echo 'y' | "$HOME/.cargo/bin/rustup" self uninstall
+    fi
+
+    # Remove cargo/rust directories
+    local rust_dirs=(
+        "$HOME/.cargo"
+        "$HOME/.rustup"
+        "$HOME/.cache/cargo"
+        "$HOME/.config/cargo"
+    )
+
+    for dir in "${rust_dirs[@]}"; do
+        if [ -d "$dir" ]; then
+            echo "🧹 Removing $dir"
+            rm -rf "$dir"
+        fi
+    done
+
+    echo "✅ Rust cleanup complete!"
+}
+
+# func: cleanup_profile
+#
+# When we uninstall rustup, we uninstall rust and cargo,
+# but to not clean the `source /home/$me/.cargo/env` reference
+# on ~/.profile, leading to useless call (and an annoying warning during login).
+#
+# This function clean this call, making a backup of it if anything goes wrong.
+cleanup_profile() {
+    echo "🐧 Creating a backup of /home/$me/.profile"
+    cp ~/.profile ~/.profile.bak
+
+    echo "🧹 Removing cargo reference from /home/$me/.profile"
+    if ! sed -i "\|source /home/$me/.cargo/env|d" ~/.profile; then
+        echo "❌ Failed to modify /home/$me/.profile. Restoring /home/$me/.profile"
+        mv ~/.profile.bak ~/.profile
+        exit 1
+    fi
+}
+
+# func: show_done
+#
+# Show some useful information after install and before start floresta node
+show_done() {
+    echo "✅ DONE"
+    echo ""
+    echo "⚠️ Before enable/start, please edit '$florestaService' to your needs. After that, run:"
+    echo ""
+    echo "    sudo systemctl start florestad.service  # this will start the service now"
+    echo "    sudo systemctl status florestad.service # this check if service is running well"
+    echo "    sudo systemctl enable florestad.service # this enable service on boot"
+    echo "    floresta-cli getblockchaininfo          # this assures all OK"
+    echo ""
+    echo "⚠️ Maybe it could be necessary to reboot your system to apply the new changes"
+    echo ""
+}
+
+install_floresta() {
+    echo "🐧 Installing $tarSrc for $me"
+    validate_network
+    apt_install gcc build-essential pkg-config libssl-dev mold
+    install_rustup
+    setup_service
+    download_floresta
+    build_floresta
+    cleanup_apt gcc build-essential pkg-config libssl-dev mold
```

Comment:
One thing we can do as a follow-up is getting a bit smarter here. Because, what if the user already has `gcc` or `build-essential` installed? We will uninstall it anyway?